/*
 * This file was automatically generated by EvoSuite
 * Fri Apr 12 01:45:17 GMT 2024
 */

package com.sonos.services._1_1;

import org.junit.Test;
import static org.junit.Assert.*;
import com.sonos.services._1_1.CustomFault;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class CustomFault_ESTest extends CustomFault_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      com.sonos.services._1.CustomFault customFault0 = new com.sonos.services._1.CustomFault();
      CustomFault customFault1 = new CustomFault("", customFault0);
      com.sonos.services._1.CustomFault customFault2 = customFault1.getFaultInfo();
      assertSame(customFault0, customFault2);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      CustomFault customFault0 = new CustomFault();
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      CustomFault customFault0 = new CustomFault("");
      com.sonos.services._1.CustomFault customFault1 = customFault0.getFaultInfo();
      assertNull(customFault1);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      CustomFault customFault0 = new CustomFault("", (com.sonos.services._1.CustomFault) null, (Throwable) null);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      CustomFault customFault0 = new CustomFault("F$O~RYaY", (Throwable) null);
  }
}
